File: procesar_contacto.php
<?php
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;

require 'path/to/PHPMailer/src/Exception.php';
require 'path/to/PHPMailer/src/PHPMailer.php';
require 'path/to/PHPMailer/src/SMTP.php';

if ($_SERVER["REQUEST_METHOD"] == "POST") {
    // Sanitizar y validar los inputs del formulario
    $nombre = filter_var($_POST['nombre'], FILTER_SANITIZE_STRING);
    $email = filter_var($_POST['email'], FILTER_VALIDATE_EMAIL);
    $mensaje = filter_var($_POST['mensaje'], FILTER_SANITIZE_STRING);

    if ($email === false) {
        echo "<div class='mensaje'><h1>Error en el email</h1>";
        echo "<p>La dirección de correo no es válida. Por favor, intenta de nuevo.</p></div>";
        exit;
    }

    // Configuración de PHPMailer
    $mail = new PHPMailer(true);
    try {
        // Configuración del servidor SMTP
        $mail->isSMTP();
        $mail->Host       = 'smtp.example.com';    // Configuración SMTP
        $mail->SMTPAuth   = true;
        $mail->Username   = 'miguel.seb.san@gmail.com';
        $mail->Password   = 'Lilmiguel03';
        $mail->SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS;
        $mail->Port       = 587;

        // Remitente y destinatarios
        $mail->setFrom($email, $nombre);
        $mail->addAddress('sseebbaassttiiaann838@gmail.com');

        // Contenido del correo
        $mail->isHTML(true);
        $mail->Subject = 'Nuevo mensaje de contacto desde el sitio web';
        $mail->Body    = "Nombre: $nombre<br>Email: $email<br>Mensaje: $mensaje";

        // Enviar el correo
        $mail->send();
        header('Location: confirmacion.html'); 
        exit;
    } catch (Exception $e) {
        echo "<div class='mensaje'><h1>Error al Enviar</h1>";
        echo "<p>Hubo un problema al enviar tu mensaje. Error: {$mail->ErrorInfo}</p></div>";
    }
}
?>

File: procesar_contacto.py
from flask import Flask, render_template, request, redirect, url_for, flash
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

app = Flask(__name__)
app.secret_key = 'supersecretkey'  # Necesario para usar flash messages

@app.route('/contacto', methods=['GET', 'POST'])
def contacto():
    if request.method == 'POST':
        # Recoger y sanitizar los datos del formulario
        nombre = request.form.get('nombre')
        email = request.form.get('email')
        mensaje = request.form.get('mensaje')
        
        if not nombre or not email or not mensaje:
            flash('Todos los campos son obligatorios')
            return redirect(url_for('contacto'))
        
        # Configurar el correo
        destinatario = "sseebbaassttiiaann838@gmail.com"
        asunto = "Nuevo mensaje de contacto desde el sitio web"
        
        contenido = f"Nombre: {nombre}\nEmail: {email}\nMensaje: {mensaje}"
        
        # Enviar el correo
        try:
            msg = MIMEMultipart()
            msg['From'] = email
            msg['To'] = destinatario
            msg['Subject'] = asunto
            
            msg.attach(MIMEText(contenido, 'plain'))
            
            # Configura el servidor de correo (ejemplo usando Gmail)
            server = smtplib.SMTP('smtp.gmail.com', 587)
            server.starttls()
            server.login('sseebbaassttiiaann838@gmail.com', 'tupassword')  # Cambia por tus credenciales
            text = msg.as_string()
            server.sendmail(email, destinatario, text)
            server.quit()
            
            return redirect(url_for('confirmacion.html'))
        except Exception as e:
            flash(f"Error al enviar el mensaje: {str(e)}")
            return redirect(url_for('contacto.html'))
    
    return render_template('contacto.html')

@app.route('confirmacion.html')
def confirmacion():
    return render_template('confirmacion.html')

if __name__ == '__main__':
    app.run(debug=True)
